### ISSUE TITLE
Could Not build Flutter Android Build : Received status code 502 from bintray

### ISSUE THREAD

**Issue by Davete0302** [flutter/flutter#95247]
**Description:**

Hi I am trying to run my flutter app debug. I last opened this app on November 3rd week but Today when I build the app debug I am facing below exception. I've been told that bintray has been sunset, but confused as how to replace/update my gradle.

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve com.onesignal:OneSignal:[3.11.2,3.99.99].
     Required by:
         project :app
      > Failed to list versions for com.onesignal:OneSignal.
         > Unable to load Maven meta-data from https://google.bintray.com/exoplayer/com/onesignal/OneSignal/maven-metadata.xml.
            > Could not get resource 'https://google.bintray.com/exoplayer/com/onesignal/OneSignal/maven-metadata.xml'.
               > Could not GET 'https://google.bintray.com/exoplayer/com/onesignal/OneSignal/maven-metadata.xml'. Received status code 502 from server: Bad Gateway

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
* BUILD FAILED in 18s
Exception: Gradle task assembleDebug failed with exit code 1
```

**Environment**

**These are my Flutter and Dart Versions:**

- Flutter 1.17.5
- Dart Dart 2.8.4

OneSignal Plugin

`onesignal_flutter: ^2.6.4`

**Anything else:**

Distribution URL in gradle.properties:

`distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip`

Project Level Gradle:

```
buildscript {
    repositories {
        google()
        mavenCentral()
//        jcenter()
        maven {
           url 'https://maven.fabric.io/public'
        }
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:4.0.1'
        classpath 'com.google.gms:google-services:3.2.1'   // new
        classpath 'io.fabric.tools:gradle:1.+'
    }

    // #jf1dt support multidex + androidX
    subprojects {
        project.configurations.all {
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'com.android.support'
                        && !details.requested.name.contains('multidex') ) {
                    details.useVersion "27.1.1"
                }

                if (details.requested.group == 'androidx.core'
                        && !details.requested.name.contains('androidx') ) {
                    details.useVersion "1.0.1"
                }
            }
        }
    }
}

allprojects {
    repositories {
        mavenCentral()
        google()
//        jcenter()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

task clean(type: Delete) {
    delete rootProject.buildDir
}
```
(crash stacktraces, as well as any other information here)

**Comment by navaronbracke** [comment]
Bintray has been discontinued. Your dependency version of OneSignal still looks for an old bintray url. You should upgrade your OneSignal dependency to a later version that no longer needs bintray.

**Comment by Davete0302** [comment]
@navaronbracke  Thanks for the reply. That is the latest version that is compatible with our app's SDK. Do I need to update my flutter version and all other dependency?

**Comment by navaronbracke** [comment]
@Davete0302 yes you will have to upgrade. You are quite a lot of major version behind, so beaware of breaking changes from Flutter 1.17 to  Flutter 2.8.

**Comment by mamuseferha** [comment]
Hi @Davete0302, this issue is likely caused because you are on an outdated version of flutter. Please upgrade your flutter version by running  `flutter upgrade`. 

If the issue persists, please provide your updated `flutter doctor -v`, your `flutter run --verbose` and a minimal reproducible code sample.
Thank you

**Comment by 78Ronin** [comment]
I managed to solve the problem as follows:
1. Check your app/build.gradle file

In my case, this file contained the following lines:
    `implementation` 'com.google.android.exoplayer: exoplayer-core: 2.15.1'
    implementation 'com.google.android.exoplayer: exoplayer-dash: 2.15.1'
    implementation` 'com.google.android.exoplayer: exoplayer-ui: 2.15.1'``
I deleted them.

2. (Optional) Update flutter to 2.8 version
3. Downgrade audioplayers to `audioplayers: ^0.16.0`
4. In app/build.gradle set targetSdkVersion and CompileSdkversuion to 30.
5. in gradle-properties file set `distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip`
6. In android/build.gradle set ext.kotlin_version = '1.4.32' and classpath 'com.android.tools.build:gradle:4.1.0'
7. Replace jcenter() to mavenCentral()

It's steps work for me. GoodLuck!
